drivers/gpio/gpio_xlnx_ps_bank.c

```diff
@@ -0,0 +1,431 @@
+{
+	const struct gpio_xlnx_ps_bank_dev_cfg *dev_conf = dev->config;
+	uint32_t				bank_data;
+
+	bank_data  = sys_read32(GPIO_XLNX_PS_BANK_DATA_REG);
+	bank_data ^= pins;
+	sys_write32(bank_data, GPIO_XLNX_PS_BANK_DATA_REG);
+
+	return 0;
+}
+
+/**
+ * @brief Configures the interrupt behaviour of a pin within the
+ *        current GPIO bank.
+ *
+ * Configures the interrupt behaviour of a pin within the current
+ * GPIO bank. If a pin is to be configured to trigger an interrupt,
+ * the following modes are supported:
+ *
+ * - edge or level triggered,
+ * - rising edge / high level or falling edge / low level,
+ * - in edge mode only: trigger on both rising and falling edge.
+ *
+ * @param dev  Pointer to the GPIO bank's device.
+ * @param pin  Index of the pin within the bank to be configured
+ *             (decimal index of the pin).
+ * @param mode Mode configuration: edge, level or interrupt disabled.
+ * @param trig Trigger condition configuration: high/low level or
+ *             rising/falling/both edge(s).
+ *
+ * @retval 0 if the interrupt configuration completed successfully,
+ *         -EINVAL if the specified pin index is out of range,
+ *         -ENOTSUP if the interrupt configuration data contains an
+ *         invalid combination of configuration flags.
+ */
+static int gpio_xlnx_ps_bank_pin_irq_configure(const struct device *dev,
+					       gpio_pin_t pin,
+					       enum gpio_int_mode mode,
+					       enum gpio_int_trig trig)
+{
+	const struct gpio_xlnx_ps_bank_dev_cfg	*dev_conf = dev->config;
+	uint32_t				pin_mask  = BIT(pin);
+	uint32_t				int_type_data;
+	uint32_t				int_polarity_data;
+	uint32_t				int_any_data;
+
+	if (pin >= dev_conf->pin_count) {
+		return -EINVAL;
+	}
+
+	int_type_data     = sys_read32(GPIO_XLNX_PS_BANK_INT_TYPE_REG);
+	int_polarity_data = sys_read32(GPIO_XLNX_PS_BANK_INT_POLARITY_REG);
+	int_any_data      = sys_read32(GPIO_XLNX_PS_BANK_INT_ANY_REG);
+
+	if (mode != GPIO_INT_MODE_DISABLED) {
+
+		if (mode == GPIO_INT_MODE_LEVEL) {
+			int_type_data &= ~pin_mask;
+		} else if (mode == GPIO_INT_MODE_EDGE) {
+			int_type_data |= pin_mask;
+		} else {
+			return -EINVAL;
+		}
+
+		if (trig == GPIO_INT_TRIG_LOW) {
+			int_polarity_data &= ~pin_mask;
+			int_any_data      &= ~pin_mask;
+		} else if (trig == GPIO_INT_TRIG_HIGH) {
+			int_polarity_data |= pin_mask;
+			int_any_data      &= ~pin_mask;
+		} else if (trig == GPIO_INT_TRIG_BOTH) {
+			if (mode == GPIO_INT_MODE_LEVEL) {
+				return -EINVAL;
+			}
+			int_any_data |= pin_mask;
+		}
+
+	} else { /* mode == GPIO_INT_MODE_DISABLED */
+		int_type_data		&= ~pin_mask;
+		int_polarity_data	&= ~pin_mask;
+		int_any_data		&= ~pin_mask;
+	}
+
+	sys_write32(int_type_data,     GPIO_XLNX_PS_BANK_INT_TYPE_REG);
+	sys_write32(int_polarity_data, GPIO_XLNX_PS_BANK_INT_POLARITY_REG);
+	sys_write32(int_any_data,      GPIO_XLNX_PS_BANK_INT_ANY_REG);
+
+	if (mode != GPIO_INT_MODE_DISABLED) {
+		sys_write32(pin_mask, GPIO_XLNX_PS_BANK_INT_EN_REG);
```

Comment:
I'm not familiar with the architecture but it seems we should write to `GPIO_XLNX_PS_BANK_INT_STAT_REG` to clear possibly dirty interrupt status bit before enabling the interrupt.

Reply:
Implemented as suggested, clearing potential stale pending interrupts makes sense.